include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+				intval(api_user()),
+				dbesc($album));
+		if (!dbm::is_result($r))
+			throw new BadRequestException("album not available");
+
+		// function for setting the items to "deleted = 1" which ensures that comments, likes etc. are not shown anymore
+		// to the user and the contacts of the users (drop_items() performs the federation of the deletion to other networks
+		foreach ($r as $rr) {
+			$photo_item = q("SELECT `id` FROM `item` WHERE `uid` = %d AND `resource-id` = '%s' AND `type` = 'photo'",
+						intval(local_user()),
+						dbesc($rr['resource-id'])
+						);
+
+			if (!dbm::is_result($photo_item))
+				throw new InternalServerErrorException("problem with deleting items occured");
+
+			drop_item($photo_item[0]['id'],false);
 		}
 
-		$r = q("SELECT `resource-id`, MAX(`scale`) AS `scale`, `album`, `filename`, `type`
-				FROM `photo`
-				WHERE `uid` = %d AND `album` != 'Contact Photos' GROUP BY `resource-id`, `album`, `filename`, `type`",
+		// now let's delete all photos from the album
+		$result = q("DELETE FROM `photo` WHERE `uid` = %d AND `album` = '%s'",
+				intval(api_user()),
+				dbesc($album));
+
+		// return success of deletion or error message
+		if ($result) {
+			$answer = array('result' => 'deleted', 'message' => 'album `' . $album . '` with all containing photos has been deleted.');
+			return api_format_data("photoalbum_delete", $type, array('$result' => $answer));
+		}
+		else {
+			throw new BadRequestException("unknown error");
+		}
+
+	}
+
+	/**
+	 * @brief update the name of the album for all photos of an album
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_fr_photoalbum_update($type) {
+		if (api_user()===false)
+			throw new ForbiddenException();
+
+		// input params
+		$album = (x($_REQUEST,'album') ? $_REQUEST['album'] : "");
+		$album_new = (x($_REQUEST,'album_new') ? $_REQUEST['album_new'] : "");
+
+		// we do not allow calls without album string
+		if ($album == "")
+			throw new BadRequestException("no albumname specified");
+		if ($album_new == "")
+			throw new BadRequestException("no new albumname specified");
+
+		// check if album is existing
+		$r = q("SELECT * FROM `photo` WHERE `uid` = %d AND `album` = '%s'",
+				intval(api_user()),
+				dbesc($album));
+		if (!dbm::is_result($r))
+			throw new BadRequestException("album not available");
+
+		// now let's update all photos to the albumname
+		$result = q("UPDATE `photo` SET `album` = '%s' WHERE `uid` = %d AND `album` = '%s'",
+				dbesc($album_new),
+				intval(api_user()),
+				dbesc($album));
+
+		// return success of updating or error message
+		if ($result) {
+			$answer = array('result' => 'updated', 'message' => 'album `' . $album . '` with all containing photos has been renamed to `' . $album_new . '`.');
+			return api_format_data("photoalbum_update", $type, array('$result' => $answer));
+		}
+		else {
+			throw new BadRequestException("unknown error");
```

Comment:
dito, code shall never reach this point as this is checked before. However, changed to "unknown error - updating in database failed".